src/Controller/PaymentController.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace App\Controller;
+
+use App\Entity\User;
+use Stripe\StripeClient;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+
+#[Route('/payment')]
+
+class PaymentController extends AbstractController
+{
+    private StripeClient $stripe;
+
+    public function __construct()
+    {
+        $this->stripe = new StripeClient($this->getParameter('stripe_secret_key'));
+    }
+
+    #[Route('/payment', name: 'app_payment')]
+    public function payment(): Response
+    {
+        $checkoutSession = $this->stripe->checkout->sessions->create([
+            'line_items' => [[
+              'price_data' => [
+                'currency' => 'eur',
+                'product_data' => [
+                  'name' => 'Abonnement Premium',
+                ],
+                'unit_amount' => 1000,
+              ],
+              'quantity' => 1,
+            ]],
+            'mode' => 'payment',
+            'success_url' => $this->getParameter('root_url') . "payment/success-url?session_id={CHECKOUT_SESSION_ID}",
+            'cancel_url' => $this->generateUrl('cancel_url', [], UrlGeneratorInterface::ABSOLUTE_URL),
+          ]);
+
+        header("HTTP/1.1 303 See Other");
```

Comment:
Tu peux supprimer les deux headers si tu fais le $this->redirect, ils font la même chose !